problem_builder/answer.py

```diff
@@ -70,16 +71,11 @@ def _get_student_id(self):
 
     @staticmethod
     def _fetch_model_object(name, student_id, course_id):
-        sql_query = '''SELECT * FROM problem_builder_answer
-                         WHERE name = %s
-                           AND student_id = %s
-                           AND (course_key = %s
-                                OR course_id = %s)'''
-        params = [name, student_id, course_id, course_id]
-        try:
-            answer = Answer.objects.raw(sql_query, params)[0]
-        except IndexError:
-            raise Answer.DoesNotExist()
+        answer = Answer.objects.get(
+            Q(name=name),
+            Q(student_id=student_id),
+            Q(course_key=course_id) | Q(course_id=course_id)
+        )
         if not answer.course_key:
             answer.course_key = answer.course_id
```

Comment:
@maxrothman Yes, that's intentional to avoid a DB write. The `course_key` attribute is set here so that we can assume it's present in other parts of the code, but it does not have to be persisted.

Value of `course_id` will be copied and persisted to `course_key` in a data migration. The migration will be faked by edX and performed in batches instead.

I was thinking of doing that migration in a separate release, but I could also add it to this PR if that doesn't complicate thinks from edX devops perspective. What works better for you?